core/merginapi.h

```diff
@@ -785,11 +804,18 @@ class MerginApi: public QObject
     //! Works only when login, password and token is set in UserAuth
     void refreshAuthToken();
 
+    /**
+     * Checks if a network error should trigger a retry attempt.
+     * \param reply Network reply to check for retryable errors
+     * \returns True if the error should trigger a retry, false otherwise
+     */
+    bool isRetryableNetworkError( QNetworkReply *reply );
+
     QNetworkRequest getDefaultRequest( bool withAuth = true );
 
     bool projectFileHasBeenUpdated( const ProjectDiff &diff );
 
-    QNetworkAccessManager mManager;
+    QNetworkAccessManager *mManager = nullptr;
```

Comment:
Any reason it is changed to pointer?

Reply:
We use a pointer to allow setting as the MockNetworkManager during tests, enabling us to simulate network connection errors.